pkg/providers/azure.go

```diff
@@ -118,34 +120,42 @@ func (sc *SetUpCmd) createAzApp() error {
 	err := bo.Retry(createApp, backoff)
 	if err != nil {
 		log.Debug(err)
-		return err
+		return "", err
 	}
 
-	return nil
+	return createdAppId, nil
 }
 
-func (sc *SetUpCmd) CreateServicePrincipal() error {
+// CreateServicePrincipal creates a service principal with the given appId
+// Returns the objectId of the created service principal
+func (az *AzClient) CreateServicePrincipal(appId string) (string, error) {
 	log.Debug("creating Azure service principal...")
 	start := time.Now()
 	log.Debug(start)
 
+	if appId == "" {
+		return "", errors.New("appId cannot be empty")
+	}
+	createdObjectId := ""
+
 	createServicePrincipal := func() error {
-		createSpCmd := exec.Command("az", "ad", "sp", "create", "--id", sc.appId, "--only-show-errors")
-		out, err := createSpCmd.CombinedOutput()
+		out, err := az.CommandRunner.RunCommand("az", "ad", "sp", "create", "--id", appId, "--only-show-errors")
 		if err != nil {
 			log.Printf("%s\n", out)
 			return err
 		}
 
 		log.Debug("checking sp was created...")
-		if sc.ServicePrincipalExists() {
-			log.Debug("Service principal created successfully!")
-			end := time.Since(start)
-			log.Debug(end)
-			return nil
+		spObjId, err := az.GetServicePrincipal(appId)
+		if err != nil {
+			return errors.New("service principal not found")
 		}
+		log.Debug("Service principal created successfully!")
```

Comment:
💡 Suggestion only please: Use case suggestion: since we create `SP` for the user, I hope we have agreement from user and informed them about the age of SP and reason it was created and how user can destruct if need be.